ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -153,19 +201,25 @@ LogSegment getNextSegment(LogSegment segment) {
   }
 
   /**
-   * @param name the name of the segment required.
-   * @return a {@link LogSegment} with {@code name} if it exists.
+   * Returns the {@link LogSegment} that is logically before the given {@code segment}.
+   * @param segment the {@link LogSegment} whose "previous" segment is required.
+   * @return the {@link LogSegment} that is logically before the given {@code segment}.
    */
-  LogSegment getSegment(String name) {
-    return segmentsByName.get(name);
+  LogSegment getPrevSegment(LogSegment segment) {
```

Comment:
Also note that the ordering of log segments generally does not matter - it matters only on startup which is once again ordered correctly due to `segmentsByName` being ordered on log segment names.